Not providing "enabled" for ingressGateway in istoctl yaml causes panic: runtime error: invalid memory address or nil pointer dereference

**Bug description**
When omitting "enabled" key for ingressGateways causes istioctl to crash with following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x28252d7]

goroutine 1 [running]:
istio.io/istio/operator/cmd/mesh.getProfileNSAndEnabledComponents(0xc000b40000, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0xc0005dd440, 0x35e55c0, 0xc000f76480)
        istio.io/istio/operator/cmd/mesh/install.go:327 +0x97
istio.io/istio/operator/cmd/mesh.runApplyCmd(0xc000f162c0, 0xc000ec115d, 0xc000d47600, 0xc00068a990, 0x0, 0x0)
        istio.io/istio/operator/cmd/mesh/install.go:155 +0x365
istio.io/istio/operator/cmd/mesh.InstallCmd.func2(0xc000f162c0, 0xc000efaba0, 0x0, 0x2, 0x0, 0x0)
        istio.io/istio/operator/cmd/mesh/install.go:122 +0x46
github.com/spf13/cobra.(*Command).execute(0xc000f162c0, 0xc000efab80, 0x2, 0x2, 0xc000f162c0, 0xc000efab80)
        github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000ec8580, 0x3, 0x3, 0xc000ec8580)
        github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
        istio.io/istio/istioctl/cmd/istioctl/main.go:36 +0x128
```


[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade



**Expected behavior**
Not including "enabled" key either produces error messages indicating it is required, or appropriate default is used (docs suggest key is not erquired)

**Steps to reproduce the bug**
Here is a minimal yaml file to use with `istioctl install -f <this yaml>.yaml`
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    ingressGateways:
      - name: istio-egressgateway
```

Adding "enabled" key avoids the crash:
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    ingressGateways:
      - name: istio-egressgateway
        enabled: true
```

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

istioctl
```
client version: 1.9.2
control plane version: 1.9.2
data plane version: 1.8.0 (49 proxies), 1.9.2 (4 proxies)
```
kubectl
```
Client Version: v1.18.0
Server Version: v1.17.17-gke.1101
```

**How was Istio installed?**
Using istioctl

**Environment where the bug was observed (cloud vendor, OS, etc)**
GKE 1.17.17-gke.1101

local environment: Ubuntu 20.04 on WSL2 on windows 10

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

--- a/istioctl/cmd/revision.go
+++ b/istioctl/cmd/revision.go
@@ -968,12 +968,12 @@ func getEnabledComponents(iops *v1alpha1.IstioOperatorSpec) []string {
 		enabledComponents = append(enabledComponents, "istiod-remote")
 	}
 	for _, gw := range iops.Components.IngressGateways {
-		if gw.Enabled.Value {
+		if util.BoolValue(gw.Enabled) {
 			enabledComponents = append(enabledComponents, fmt.Sprintf("ingress:%s", gw.GetName()))
 		}
 	}
 	for _, gw := range iops.Components.EgressGateways {
-		if gw.Enabled.Value {
+		if util.BoolValue(gw.Enabled) {
 			enabledComponents = append(enabledComponents, fmt.Sprintf("egress:%s", gw.GetName()))
 		}
 	}

--- a/operator/cmd/mesh/install.go
+++ b/operator/cmd/mesh/install.go
@@ -327,13 +327,13 @@ func getProfileNSAndEnabledComponents(iop *v1alpha12.IstioOperator) (string, str
 			}
 		}
 		for _, c := range iop.Spec.Components.IngressGateways {
-			if c.Enabled.Value {
+			if util.BoolValue(c.Enabled) {
 				enabledComponents = append(enabledComponents, name.UserFacingComponentName(name.IngressComponentName))
 				break
 			}
 		}
 		for _, c := range iop.Spec.Components.EgressGateways {
-			if c.Enabled.Value {
+			if util.BoolValue(c.Enabled) {
 				enabledComponents = append(enabledComponents, name.UserFacingComponentName(name.EgressComponentName))
 				break
 			}

--- a/operator/pkg/component/component.go
+++ b/operator/pkg/component/component.go
@@ -31,6 +31,7 @@ import (
 	"istio.io/istio/operator/pkg/patch"
 	"istio.io/istio/operator/pkg/tpath"
 	"istio.io/istio/operator/pkg/translate"
+	"istio.io/istio/operator/pkg/util"
 	"istio.io/pkg/log"
 )
 
@@ -332,7 +333,7 @@ func (c *IngressComponent) Namespace() string {
 // Enabled implements the IstioComponent interface.
 func (c *IngressComponent) Enabled() bool {
 	// type assert is guaranteed to work in this context.
-	return boolValue(c.componentSpec.(*v1alpha1.GatewaySpec).Enabled)
+	return util.BoolValue(c.componentSpec.(*v1alpha1.GatewaySpec).Enabled)
 }
 
 // EgressComponent is the egress gateway component.
@@ -382,7 +383,7 @@ func (c *EgressComponent) Namespace() string {
 // Enabled implements the IstioComponent interface.
 func (c *EgressComponent) Enabled() bool {
 	// type assert is guaranteed to work in this context.
-	return boolValue(c.componentSpec.(*v1alpha1.GatewaySpec).Enabled)
+	return util.BoolValue(c.componentSpec.(*v1alpha1.GatewaySpec).Enabled)
 }
 
 // runComponent performs startup tasks for the component defined by the given CommonComponentFields.
@@ -491,12 +492,4 @@ func disabledYAMLStr(componentName name.ComponentName, resourceName string) stri
 		fullName += " " + resourceName
 	}
 	return fmt.Sprintf("%s %s %s\n", yamlCommentStr, fullName, componentDisabledStr)
-}
-
-// boolValue returns true is v is not nil and v.Value is true, or false otherwise.
-func boolValue(v *v1alpha1.BoolValueForPB) bool {
-	if v == nil {
-		return false
-	}
-	return v.Value
 }

--- a/operator/pkg/util/util.go
+++ b/operator/pkg/util/util.go
@@ -23,6 +23,8 @@ import (
 	"path/filepath"
 	"strconv"
 	"strings"
+
+	"istio.io/api/operator/v1alpha1"
 )
 
 type FileFilter func(fileName string) bool
@@ -139,4 +141,12 @@ func RenderTemplate(tmpl string, ts interface{}) (string, error) {
 		return "", err
 	}
 	return buf.String(), nil
+}
+
+// BoolValue returns true if v is not nil and v.Value is true, or false otherwise.
+func BoolValue(v *v1alpha1.BoolValueForPB) bool {
+	if v == nil {
+		return false
+	}
+	return v.Value
 }

--- a/operator/pkg/util/util_test.go
+++ b/operator/pkg/util/util_test.go
@@ -17,6 +17,10 @@ package util
 import (
 	"errors"
 	"testing"
+
+	protobuf "github.com/gogo/protobuf/types"
+
+	"istio.io/api/operator/v1alpha1"
 )
 
 func TestParseValue(t *testing.T) {
@@ -246,6 +250,42 @@ func TestRenderTemplate(t *testing.T) {
 
 			if (err == nil && tt.err != nil) || (err != nil && tt.err == nil) {
 				t.Errorf("%s: got error :%v, wanted error: %v", tt.desc, err, tt.err)
+			}
+		})
+	}
+}
+
+func TestBoolValue(t *testing.T) {
+	cases := []struct {
+		desc string
+		in   *v1alpha1.BoolValueForPB
+		out  bool
+	}{
+		{
+			desc: "nil value",
+			in:   nil,
+			out:  false,
+		},
+		{
+			desc: "false value",
+			in: &v1alpha1.BoolValueForPB{
+				BoolValue: protobuf.BoolValue{Value: false},
+			},
+			out: false,
+		},
+		{
+			desc: "true value",
+			in: &v1alpha1.BoolValueForPB{
+				BoolValue: protobuf.BoolValue{Value: true},
+			},
+			out: true,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			res := BoolValue(c.in)
+			if res != c.out {
+				t.Fatalf("%s: got %v, expected: %v", c.desc, res, c.out)
 			}
 		})
 	}

